### PR TITLE
Completed api-registration

### PR DESCRIPTION
**Pull request by K-Bowler**
api-registration

### CODE REVIEW

File: api-registration/server/index.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      const sql = `
+      insert into "users"("username", "hashedPassword")
+      values($1, $2)
+      returning *;
+      `;
+      const params = [username, password];
```
Comment (scbowler) [inline api-registration/server/index.js]: Here we want to be using the `hashedPassword`, the hashed password is what we want to save in the database.

File: api-registration/server/index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      db.query(sql, params)
+        .then(result => {
+          const newUser = result.rows[0];
+          res.status(201).json(newUser);
```
Comment (scbowler) [inline api-registration/server/index.js]: You don't want to send the password back to the client, even if it is hashed. The best way to control the data being sent back to the client is to specify the desired columns in the `returning` clause of the SQL statement.
```SQL
returning "userId", "username", "createdAt"
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, but a couple updates needed. Let me know if you have any questions.

File: api-registration/server/index.js (hunk 2 above)
Comment (K-Bowler) [reply api-registration/server/index.js]: image

File: api-registration/server/index.js (hunk 1 above)
Comment (scbowler) [reply api-registration/server/index.js]: Missed this one...

**Review by scbowler — CHANGES_REQUESTED** [review]

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Pull request merged**